pkg/compactor/blocks_cleaner.go

```diff
@@ -383,39 +400,51 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userID string, firstRun b
 }
 
 // cleanUserPartialBlocks delete partial blocks which are safe to be deleted. The provided partials map
-// is updated accordingly.
+// and index are updated accordingly.
 func (c *BlocksCleaner) cleanUserPartialBlocks(ctx context.Context, partials map[ulid.ULID]error, idx *bucketindex.Index, userBucket objstore.InstrumentedBucket, userLogger log.Logger) {
+	// Collect all blocks with missing meta.json into buffered channel.
+	blocks := make([]interface{}, 0, len(partials))
```

Comment:
Any reason we are using `[]interface{}` rather than `[]ulid`?

Reply:
FYI:https://github.com/cortexproject/cortex/blob/8cf25832319e4261df2dacdf1ff789f5c00c72de/pkg/util/concurrency/runner.go#L65